I get a test failure on my computer:
```
$ pytest sympy/printing/pretty/tests/test_pretty.py::test_issue_18272
========================================== test session starts ==========================================
platform linux -- Python 3.6.9, pytest-5.0.1, py-1.8.0, pluggy-0.12.0
architecture: 64-bit
cache:        yes
ground types: gmpy 2.0.8

rootdir: /space/enojb/sympy/sympy.git, inifile: pytest.ini
plugins: forked-1.0.2, xdist-1.29.0, instafail-0.4.1
collected 1 item                                                                                        

sympy/printing/pretty/tests/test_pretty.py F                                                      [100%]

=============================================== FAILURES ================================================
___________________________________________ test_issue_18272 ____________________________________________

    def test_issue_18272():
        from sympy.solvers import solveset
        from sympy import pretty, ConditionSet
        x = Symbol('x')
        n = Symbol('n')
    
>       assert pretty(solveset(exp(x) - x, x)) == \
        '⎧            ⎛      x    ⎞⎫\n'\
        '⎨x | x ∊ ℂ ∧ ⎝-x + ℯ  = 0⎠⎬\n'\
        '⎩                         ⎭'
E       AssertionError: assert '            ... -x + e  = 0}' == '⎧            ...            ⎭'
E         -                                 x     
E         - {x | x in S.Complexes and -x + e  = 0}
E         + ⎧            ⎛      x    ⎞⎫
E         + ⎨x | x ∊ ℂ ∧ ⎝-x + ℯ  = 0⎠⎬
E         + ⎩                         ⎭

sympy/printing/pretty/tests/test_pretty.py:6944: AssertionError
                                            DO *NOT* COMMIT!                                            
======================================= 1 failed in 2.49 seconds ========================================
```
This is happening in the optional dependency tests so it might depend which other packages you have installed or on the environment where you run the tests.

--- a/sympy/printing/pretty/pretty.py
+++ b/sympy/printing/pretty/pretty.py
@@ -2058,7 +2058,7 @@ class PrettyPrinter(Printer):
         else:
             cond = self._print(ts.condition)
             if self._use_unicode:
-                cond = self._print_seq(cond, "(", ")")
+                cond = prettyForm(*cond.parens("(", ")", ifascii_nougly=True))
 
         bar = self._print("|")
 

--- a/sympy/printing/pretty/pretty.py
+++ b/sympy/printing/pretty/pretty.py
@@ -2058,7 +2058,8 @@ class PrettyPrinter(Printer):
         else:
             cond = self._print(ts.condition)
             if self._use_unicode:
-                cond = prettyForm(*cond.parens("(", ")", ifascii_nougly=True))
+                cond = self._print(cond)
+                cond = prettyForm(*cond.parens())
 
         bar = self._print("|")
 

--- a/sympy/printing/pretty/tests/test_pretty.py
+++ b/sympy/printing/pretty/tests/test_pretty.py
@@ -6934,3 +6934,31 @@ def test_is_combining():
 def test_issue_17857():
     assert pretty(Range(-oo, oo)) == '{..., -1, 0, 1, ...}'
     assert pretty(Range(oo, -oo, -1)) == '{..., 1, 0, -1, ...}'
+
+def test_issue_18272():
+    from sympy.solvers import solveset
+    from sympy import pretty, ConditionSet
+    x = Symbol('x')
+    n = Symbol('n')
+
+    assert pretty(solveset(exp(x) - x, x)) == \
+    '⎧            ⎛      x    ⎞⎫\n'\
+    '⎨x | x ∊ ℂ ∧ ⎝-x + ℯ  = 0⎠⎬\n'\
+    '⎩                         ⎭'
+    assert pretty(solveset(Abs(2*x) - n, x, S.Reals)) == \
+    '⎧        ⎧-n   n⎫   ⎛n         ⎞⎫\n'\
+    '⎨x | x ∊ ⎨───, ─⎬ ∧ ⎜─ ∈ [0, ∞)⎟⎬\n'\
+    '⎩        ⎩ 2   2⎭   ⎝2         ⎠⎭'
+    assert pretty(ConditionSet(x, Eq(Piecewise((1, x >= 3), (x/2 - 1/2, x >= 2), (1/2, x >= 1),
+                (x/2, True)) - 1/2, 0), Interval(0, 3))) == \
+    '⎧                 ⎛⎛⎧   1     for x ≥ 3⎞          ⎞⎫\n'\
+    '⎪                 ⎜⎜⎪                  ⎟          ⎟⎪\n'\
+    '⎪                 ⎜⎜⎪x                 ⎟          ⎟⎪\n'\
+    '⎪                 ⎜⎜⎪─ - 0.5  for x ≥ 2⎟          ⎟⎪\n'\
+    '⎪                 ⎜⎜⎪2                 ⎟          ⎟⎪\n'\
+    '⎨x | x ∊ [0, 3] ∧ ⎜⎜⎨                  ⎟ - 0.5 = 0⎟⎬\n'\
+    '⎪                 ⎜⎜⎪  0.5    for x ≥ 1⎟          ⎟⎪\n'\
+    '⎪                 ⎜⎜⎪                  ⎟          ⎟⎪\n'\
+    '⎪                 ⎜⎜⎪   x              ⎟          ⎟⎪\n'\
+    '⎪                 ⎜⎜⎪   ─     otherwise⎟          ⎟⎪\n'\
+    '⎩                 ⎝⎝⎩   2              ⎠          ⎠⎭'

--- a/sympy/printing/pretty/tests/test_pretty.py
+++ b/sympy/printing/pretty/tests/test_pretty.py
@@ -6941,15 +6941,15 @@ def test_issue_18272():
     x = Symbol('x')
     n = Symbol('n')
 
-    assert xpretty(solveset(exp(x) - x, x)) == \
+    assert upretty(solveset(exp(x) - x, x)) == \
     '⎧            ⎛      x    ⎞⎫\n'\
     '⎨x | x ∊ ℂ ∧ ⎝-x + ℯ  = 0⎠⎬\n'\
     '⎩                         ⎭'
-    assert xpretty(solveset(Abs(2*x) - n, x, S.Reals)) == \
+    assert upretty(solveset(Abs(2*x) - n, x, S.Reals)) == \
     '⎧        ⎧-n   n⎫   ⎛n         ⎞⎫\n'\
     '⎨x | x ∊ ⎨───, ─⎬ ∧ ⎜─ ∈ [0, ∞)⎟⎬\n'\
     '⎩        ⎩ 2   2⎭   ⎝2         ⎠⎭'
-    assert xpretty(ConditionSet(x, Eq(Piecewise((1, x >= 3), (x/2 - 1/2, x >= 2), (1/2, x >= 1),
+    assert upretty(ConditionSet(x, Eq(Piecewise((1, x >= 3), (x/2 - 1/2, x >= 2), (1/2, x >= 1),
                 (x/2, True)) - 1/2, 0), Interval(0, 3))) == \
     '⎧                 ⎛⎛⎧   1     for x ≥ 3⎞          ⎞⎫\n'\
     '⎪                 ⎜⎜⎪                  ⎟          ⎟⎪\n'\

--- a/sympy/printing/pretty/tests/test_pretty.py
+++ b/sympy/printing/pretty/tests/test_pretty.py
@@ -32,9 +32,11 @@ from sympy.physics import mechanics
 from sympy.physics.units import joule, degree
 from sympy.printing.pretty import pprint, pretty as xpretty
 from sympy.printing.pretty.pretty_symbology import center_accent, is_combining
+from sympy import ConditionSet
 
 from sympy.sets import ImageSet, ProductSet
 from sympy.sets.setexpr import SetExpr
+from sympy.solvers import solveset
 from sympy.tensor.array import (ImmutableDenseNDimArray, ImmutableSparseNDimArray,
                                 MutableDenseNDimArray, MutableSparseNDimArray, tensorproduct)
 from sympy.tensor.functions import TensorProduct
@@ -6936,20 +6938,18 @@ def test_issue_17857():
     assert pretty(Range(oo, -oo, -1)) == '{..., 1, 0, -1, ...}'
 
 def test_issue_18272():
-    from sympy.solvers import solveset
-    from sympy import pretty, ConditionSet
     x = Symbol('x')
     n = Symbol('n')
 
-    assert pretty(solveset(exp(x) - x, x)) == \
+    assert xpretty(solveset(exp(x) - x, x)) == \
     '⎧            ⎛      x    ⎞⎫\n'\
     '⎨x | x ∊ ℂ ∧ ⎝-x + ℯ  = 0⎠⎬\n'\
     '⎩                         ⎭'
-    assert pretty(solveset(Abs(2*x) - n, x, S.Reals)) == \
+    assert xpretty(solveset(Abs(2*x) - n, x, S.Reals)) == \
     '⎧        ⎧-n   n⎫   ⎛n         ⎞⎫\n'\
     '⎨x | x ∊ ⎨───, ─⎬ ∧ ⎜─ ∈ [0, ∞)⎟⎬\n'\
     '⎩        ⎩ 2   2⎭   ⎝2         ⎠⎭'
-    assert pretty(ConditionSet(x, Eq(Piecewise((1, x >= 3), (x/2 - 1/2, x >= 2), (1/2, x >= 1),
+    assert xpretty(ConditionSet(x, Eq(Piecewise((1, x >= 3), (x/2 - 1/2, x >= 2), (1/2, x >= 1),
                 (x/2, True)) - 1/2, 0), Interval(0, 3))) == \
     '⎧                 ⎛⎛⎧   1     for x ≥ 3⎞          ⎞⎫\n'\
     '⎪                 ⎜⎜⎪                  ⎟          ⎟⎪\n'\

